pkg/annotations/ingress_annotations.go

```diff
@@ -274,12 +274,21 @@ func BackendProtocol(ing *networking.Ingress) (ProtocolEnum, error) {
 func GetHostNameExtensions(ing *networking.Ingress) ([]string, error) {
 	val, err := parseString(ing, HostNameExtensionKey)
 	if err == nil {
-		var hostnames []string
+		// use a map to enforce uniqueness
+		uniqueHostnames := make(map[string]interface{})
 		for _, hostname := range strings.Split(val, ",") {
-			if len(hostname) > 0 {
-				hostnames = append(hostnames, strings.TrimSpace(hostname))
+			trimmed := strings.TrimSpace(hostname)
+			if len(trimmed) > 0 {
+				uniqueHostnames[trimmed] = nil
 			}
 		}
+
+		// convert the map to a slice
+		var hostnames []string
+		for key := range uniqueHostnames {
```

Comment:
Go `map` is unordered and will lose the ordering that customer gave.

Reply:
Ah my bad for not realizing the order was significant, I've changed it to preserve the order